include/pinocchio/multibody/model-graph.hpp

```diff
@@ -0,0 +1,794 @@
+//
+// Copyright (c) 2025 INRIA
+//
+
+#ifndef __pinocchio_multibody_model_graph_hpp__
+#define __pinocchio_multibody_model_graph_hpp__
+
+#include "pinocchio/multibody/fwd.hpp"
+#include "pinocchio/multibody/joint/joint-generic.hpp"
+#include "pinocchio/multibody/model.hpp"
+#include "pinocchio/spatial/inertia.hpp"
+#include "pinocchio/spatial/se3.hpp"
+
+#include <Eigen/Core>
+
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/depth_first_search.hpp>
+#include <boost/graph/visitors.hpp>
+
+#include <vector>
+#include <string>
+#include <iostream>
+#include <unordered_map>
+
+namespace pinocchio
+{
+  struct JointFixedGraph
+  {
+    JointFixedGraph() = default;
+  };
+
+  struct JointRevoluteGraph
+  {
+    // rotation axis
+    Eigen::Vector3d axis;
+
+    JointRevoluteGraph(const Eigen::Vector3d & ax)
+    : axis(ax)
+    {
+    }
+  };
+
+  struct JointPrismaticGraph
+  {
+    Eigen::Vector3d axis;
+    JointPrismaticGraph(const Eigen::Vector3d & ax)
+    : axis(ax)
+    {
+    }
+  };
+
+  struct JointFreeFlyerGraph
+  {
+    JointFreeFlyerGraph() = default;
+  };
+
+  struct JointSphericalGraph
+  {
+    JointSphericalGraph() = default;
+  };
+
+  // Flipped whne model is reversed ?
+  struct JointSphericalZYXGraph
+  {
+    JointSphericalZYXGraph() = default;
+  };
+
+  struct JointTranslationGraph
+  {
+    JointTranslationGraph() = default;
+  };
+
+  struct JointPlanarGraph
+  {
+    JointPlanarGraph() = default;
+  };
+
+  struct JointHelicalGraph
+  {
+    Eigen::Vector3d axis;
+    double pitch;
+
+    JointHelicalGraph(const Eigen::Vector3d & ax, const double & p)
+    : axis(ax)
+    , pitch(p) {};
+  };
+
+  struct JointUniversalGraph
+  {
+    Eigen::Vector3d axis1;
+    Eigen::Vector3d axis2;
+
+    JointUniversalGraph(const Eigen::Vector3d & ax1, const Eigen::Vector3d & ax2)
+    : axis1(ax1)
+    , axis2(ax2) {};
+  };
+
+  // Forward declare
+  struct JointCompositeGraph;
+
+  using JointGraphVariant = boost::variant<
+    JointFixedGraph,
+    JointRevoluteGraph,
+    JointPrismaticGraph,
+    JointFreeFlyerGraph,
+    JointSphericalGraph,
+    JointSphericalZYXGraph,
+    JointTranslationGraph,
+    JointPlanarGraph,
+    JointHelicalGraph,
+    JointUniversalGraph,
+    boost::recursive_wrapper<JointCompositeGraph>>;
+
+  struct JointCompositeGraph
+  {
+    std::vector<JointGraphVariant> joints;
+    std::vector<SE3> jointsPlacements;
+
+    JointCompositeGraph() = default;
+
+    JointCompositeGraph(const JointGraphVariant & j, const SE3 jPose)
+    {
+      joints.push_back(j);
+      jointsPlacements.push_back(jPose);
+    };
```

Comment:
```suggestion
    }
```

Pas besoin de ; après une définition (à supprimer partout)